packages/data/src/world/mq/shadow_temple_mq.yml

```diff
@@ -237,7 +238,7 @@
   dungeon: Shadow
   exits:
     "Shadow Temple Lower Huge Pit Ledge Before Invisible Spike Floors": "small_keys_shadow(3)"
-    "Shadow Temple Invisible Spike Floors Room Upper Ledge Before Door": "(event(SHADOW_MQ_INVISIBLE_SPIKE_FLOORS_ICE_PLATFORMS) && (can_longshot || (has_tunic_goron_strict && can_hookshot))) || climb_anywhere || hookshot_anywhere" #Goron Tunic acts on the spikes as well, allowing you to use hookshot while standing on the spikes.
+    "Shadow Temple Invisible Spike Floors Room Upper Ledge Before Door": "hookshot_anywhere || climb_anywhere || (((is_adult && can_hookshot && (event(SHADOW_MQ_INVISIBLE_SPIKE_FLOORS_ICE_PLATFORMS) || event(SHADOW_MQ_INVISIBLE_SPIKE_FLOORS_REDEADS_CLEAR))) || can_longshot) && (glitch_megaflip || has_hover_boots))" #Worst-case scenario because the ice platforms can block access to the door.
```

Comment:
```suggestion
    "Shadow Temple Invisible Spike Floors Room Upper Ledge Before Door": "hookshot_anywhere || climb_anywhere || (((is_adult && can_hookshot && (event(SHADOW_MQ_INVISIBLE_SPIKE_FLOORS_ICE_PLATFORMS) || event(SHADOW_MQ_INVISIBLE_SPIKE_FLOORS_REDEADS_CLEAR))) || can_longshot) && (glitch_megaflip || has_hover_boots || can_jump_slash))" #Worst-case scenario because the ice platforms can block access to the door.
```